bndtools.utils/src/org/bndtools/utils/javaproject/JavaProjectUtils.java

```diff
@@ -106,8 +77,8 @@ public static String getJavaLevel(IJavaProject project) throws Exception {
 					continue;
 				if (PATH_JRE_CONTAINER.equals(path.segment(0)) && PATH_JRE_STANDARD_VM.equals(path.segment(1))) {
 					String jreName = path.segment(2);
-					return JavaLevel.fromLabel(jreName)
-						.getLevel();
+					return EE.parse(jreName)
```

Comment:
There is a slight behavioral difference here. `JavaLevel.fromLabel` would throw an IllegalArgumentException if the jre name was unknown. `EE.parse` will return null which will lead to an NPE when `getVersionLabel()` is called. So I think you need to check for null from `EE.parse` and throw an IllegalArgument exception then to preserve the semantics of this method.

```java
EE ee = EE.parse(jreName);
if (ee == null) {
  throw new IllegalArgumentException("Unrecognized Java compliance level: " + jreName);
}
return ee.getVersionLabel();
```